### PR TITLE
Add initial dev docs describing the overall design

### PR DESCRIPTION
**Pull request by komish**
This PR just adds a high-level developer documentation to clarify the overall design goals. This does make reference to existing PRs that have not merged for the moment, but will be updated after the fact if those PRs change, etc. 

Signed-off-by: Jose R. Gonzalez <josegonzalez89@gmail.com>

### CODE REVIEW

File: docs/dev/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+## CLI Implementation
+
+The Preflight CLI utilizes the libraries and design mentioned above. The bulk
+of the CLI code is found in the `cmd` package.
+
+Currently, we assume that the user must provide a single positional argument:
+the container image which will be checked for compliance with our policies.
+
+The user can then further tailor their execution by specifying configurables
+such as the exact policies to enforce, and the output format. The CLI current
```
Comment (bcrochet) [inline docs/dev/README.md]: "currently"

**Review by bcrochet — COMMENTED** [review]
One small nit. And perhaps a 'prereq' section would be good? Would contain 'podman' for instance?

File: docs/dev/README.md (hunk 1 above)
Comment (komish) [reply docs/dev/README.md]: Fixed 👍

**Comment by komish** [comment]
> And perhaps a 'prereq' section would be good? Would contain 'podman' for instance?

Definitely worth adding, though I think it lends itself more to user docs than dev docs at the moment. As in, I think our initial implementation will use podman under the hood, and therefore the user executing our CLI will need to know about it. If it's acceptable, I can add it at a later time after we've fully implemented all the various checks.

**Review by bcrochet — APPROVED** [review]

**Pull request merged**